v0/destinations/posthog/transform.js

```diff
@@ -0,0 +1,105 @@
+    { destKey: "$host", sourceKeys: "" },
+    { destKey: "$pathname", sourceKeys: "context.page.path" },
+    { destKey: "$browser_version", sourceKeys: "context.browser.version" },
+    { destKey: "$screen_height", sourceKeys: "context.screen.height" },
+    { destKey: "$screen_width", sourceKeys: "context.screen.width" },
+    { destKey: "$lib", sourceKeys: "context.library.name" },
+    { destKey: "$lib_version", sourceKeys: "context.library.version" },
+    { destKey: "$insert_id", sourceKeys: "" }, // TO DO : Need to Check
+    { destKey: "$time", sourceKeys: ["originalTimestamp", "timestamp"] },
+    { destKey: "$device_id", sourceKeys: "context.device.id" },
+    { destKey: "distinct_id", sourceKeys: "userId" },
+    { destKey: "$initial_referrer", sourceKeys: "properties.referrer" },
+    { destKey: "$initial_referring_domain", sourceKeys: "properties.url" },
+    { destKey: "$ip", sourceKeys: "context.ip" },
+    { destKey: "$timestamp", sourceKeys: ["originalTimestamp", "timestamp"] },
+    { destKey: "$anon_distinct_id", sourceKeys: "anonymousId" }
+  ];
+  // TO DO :: Delete Property that is already defined in above json
+  const data = constructPayload(message, PHPropertyJson);
+
+  return data;
+};
+
+const responseBuilderSimple = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    // eslint-disable-next-line no-undef
+    throw Error(Message.FailedToConstructPayload);
+  }
+
+  if (payload.properties) {
+    payload.properties = {
+      ...generatePropertyDefination(message),
+      ...payload.properties
+    };
+  }
+
+  // Mapping Destination Event with correct value
+  if (category.type !== CONFIG_CATEGORIES.TRACK.type) {
+    payload.event = category.event;
+  }
+
+  const responseBody = {
+    ...payload,
+    api_key: destination.Config.teamApiKey,
+    type:
+      category.type === CONFIG_CATEGORIES.TRACK.type ? "capture" : category.type // To Do:: Need to improve this line of code. Figure out some other ways.
+  };
+  const response = defaultRequestConfig();
+  response.endpoint = `${destination.Config.yourInstance}/batch`;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Indicative-Client": "RudderStack",
+    "Content-Type": "application/json"
+  };
+  response.userId = getFieldValueFromMessage(message, "userId");
+  response.body.JSON = responseBody;
+  return response;
+};
+
+// Validate Message Type coming from source
+const validateMessageType = message => {
+  if (!message.type) {
+    throw Error(Message.TypeNotFound);
+  }
+
+  const category = CONFIG_CATEGORIES[message.type.toUpperCase()];
+  if (!category) {
+    throw Error(Message.TypeNotSupported);
+  }
+};
+
+const processEvent = (message, destination) => {
+  validateMessageType(message);
+
+  const messageType = message.type.toLowerCase();
+
+  const repList = [];
```

Comment:
There is a one-one mapping between input and output events. We don't need to wrap it in an array.

Reply:
Fixed